src/olympia/urls.py

```diff
@@ -20,6 +21,22 @@
 handler404 = 'olympia.amo.views.handler404'
 handler500 = 'olympia.amo.views.handler500'
 
+
+def serve_static_files(request, path, **kwargs):
+    if settings.TARGET == 'production':
+        return serve_static(request, path, document_root=settings.STATIC_ROOT, **kwargs)
+    else:
+        return static_serve(request, path, insecure=True, show_indexes=True, **kwargs)
+
+
+def serve_javascript_catalog(request, locale, **kwargs):
+    with translation.override(locale):
+        return JavaScriptCatalog.as_view()(request, locale, **kwargs)
```

Comment:
I don't think we want these to be available on "real" environment - so why not use the same trick as for FxA and only make them available if `ENV == 'local'` ? Don't need to move the code around, can just replace the existing condition

Reply:
I tried that, but got failing tests. My thought was, it's not actually a problem to serve these paths as they effectively cannot be reached in production. and even if they were, that means CDN and NGINX could not serve the file... so wouldn't serving it here actually be an improvement?

If you feel strongly I'll add a gate but it would have to be that we register the URLS and then 404 if env !== local.

When testing this with the gate at the register level the "override_settings" is applied too late so it doesn't behave as expected unless you pass the environment variable when executing the test. That felt like overkill...